gdk/commands/component/BuildCommand.py

```diff
@@ -159,42 +161,10 @@ def run_build_command(self):
             raise Exception(f"Error building the component with the given build system.\n{e}")
 
     def _build_system_zip(self):
-        """
-        Builds the component as a zip file.
-
-        Copies over necessary files by excluding certain files to a build folder identied for zip build system
-        (supported_component_builds.json has the build folder info).
-        This build folder is zipped completely as a component zip artifact.
-        Raises an exception if there's an error in the process of zippings.
-
-        Parameters
-        ----------
-            None
-
-        Returns
-        -------
-            None
-        """
-        try:
-            zip_build = next(iter(self._get_build_folder_by_build_system()))  # Only one zip-build folder in the set
-            artifacts_zip_build = Path(zip_build).joinpath(utils.current_directory.name).resolve()
-            utils.clean_dir(zip_build)
-            logging.debug("Copying over component files to the '{}' folder.".format(artifacts_zip_build.name))
-            shutil.copytree(utils.current_directory, artifacts_zip_build, ignore=self._ignore_files_during_zip)
-
-            # Get build file name without extension. This will be used as name of the archive.
-            archive_file = utils.current_directory.name
-            logging.debug(
-                "Creating an archive named '{}.zip' in '{}' folder with the files in '{}' folder.".format(
-                    archive_file, zip_build.name, artifacts_zip_build.name
-                )
-            )
-            archive_file_name = Path(zip_build).joinpath(archive_file).resolve()
-            shutil.make_archive(archive_file_name, "zip", root_dir=artifacts_zip_build)
-            logging.debug("Archive complete.")
-
-        except Exception as e:
-            raise Exception("""Failed to zip the component in default build mode.\n{}""".format(e))
+        # Delegate to avoid breaking tests - TODO: We need to avoid testing private methods
+        build_system = BuildSystem()
+        build_system.register(Zip(self.project_config, self._get_build_folder_by_build_system()))
```

Comment:
For later PRs if needed: Should we register all the supported build systems during the Build command initialization (similar to this `self.supported_build_sytems = project_utils.get_supported_component_builds() in L20`)

And call this `build_system.build("zip")` in L 153 and/or using the `self.project_config["component_build_config"]["build_system"]` instead of "zip"

Reply:
Yeah we should do that. For now this is just minimizing the surface area of the changes but you are right